### DISCUSSION TITLE
VlogCam Mobile app

### DISCUSSION THREAD

**Discussion by shreemaan-abhishek** [scaleracademy/open-source-projects#41] [Ideas]
## Summary

VlogCam is a video recorder app that lets you switch between cameras (rear and front) while you're recording a video. This feature is inbuilt in some high-end devices like Samsung S10, VlogCam will support this on all devices.

## Assumption 

A device that's running this app should probably have powerful hardware as achieving this would be a resource expensive work.

## How can this be done?

1. Record the video taking input from both the cameras, keep track of when did the user switched onto which camera. Then use that information to filter out unnecessary portions of the video from both the cameras and then merge the videos.
2. Use the volume buttons to switch between active cameras, keep the screen free from buttons or any other views, then record the screen. Sounds unrealistic but could be very eco-friendly on hardware.
2. Create different bits of videos (recorded using the active camera) and merge them finally.

**Comment by championswimmer** [comment]
Most mid to low end phones do not allow recording multiple camera streams simultaneously on the device due to constraints on the SoC image processing bandwidth :(
